From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zljing <zljing@codeaurora.org>
Date: Wed, 2 Aug 2017 15:10:40 +0800
Subject: [PATCH 063/226] ShutdownThread: Shorten wait time for shutdown time
 optimization

Default 500ms delay is waitting shutdown actions done in while cycle.
Shutdown actions include broadcast,nfc,radio,wifi and mountservice.
The whole actions are ordered serial operations. This delay is too
long for broadcast/nfc/radio/wifi/mountservice shutdown. Decrease the
delay will add mActionDoneSync lock time and while cycle times, so this
delay also can't be too short. Set half will have 0.5~0.7s benefit for
shutdown time, total time from 2.4s to 1.7~1.9s, which is comparable
with ref device.

CRs-Fixed: 2023578

Change-Id: Ieb34034652c3cba1c41eadc00500d1f211c2a009
---
 services/core/java/com/android/server/power/ShutdownThread.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/power/ShutdownThread.java b/services/core/java/com/android/server/power/ShutdownThread.java
index ec9e276c31..0587bfba37 100644
--- a/services/core/java/com/android/server/power/ShutdownThread.java
+++ b/services/core/java/com/android/server/power/ShutdownThread.java
@@ -88,7 +88,7 @@ public final class ShutdownThread extends Thread {
     private static final int MOUNT_SERVICE_STOP_PERCENT = 20;
 
     // length of vibration before shutting down
-    @VisibleForTesting static final int DEFAULT_SHUTDOWN_VIBRATE_MS = 500;
+    @VisibleForTesting static final int DEFAULT_SHUTDOWN_VIBRATE_MS = 250;
 
     // state tracking
     private static final Object sIsStartedGuard = new Object();
